src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -455,16 +464,39 @@ func (c controller) postInstallConfigs(ctx context.Context) error {
 		return errors.Wrapf(err, "Timeout while waiting for console to become available")
 	}
 
+	// Wait for OLM operators
+	if err = c.waitForOLMOperators(ctx); err != nil {
+		return errors.Wrapf(err, "Error while initializing OLM operators")
+	}
+
+	return nil
+}
+
+func (c controller) waitForOLMOperators(ctx context.Context) error {
+	// Get maximum wait timeout for OLM operators:
+	operators, err := c.ic.GetClusterMonitoredOLMOperators(context.TODO(), c.ClusterID)
+	if err != nil {
+		return errors.Wrapf(err, "Error while fetch the monitored operators from assisted-service.")
+	}
+	waitTimeout := c.getMaximumOLMTimeout(operators)
+	c.log.Infof("OLM operators wait timeout %v", waitTimeout)
+
+	// Wait for the CSV state of the OLM operators, before applying OLM CRs
+	err = utils.WaitForPredicateParamsWithContext(ctx, waitTimeout, GeneralWaitInterval, c.waitForCSVBeCreated, operators)
+	if err != nil {
+		c.log.WithError(err).Warnf("Failed to wait for some of the OLM operators to be initilized")
```

Comment:
can you please comment why we continue in case of failure?

Reply:
Done.